autoremovetorrents/filter/path.py

```diff
@@ -0,0 +1,33 @@
+#-*- coding:utf-8 -*-
+
+from .filter import Filter
+
+class PathFilter(Filter):
+    def __init__(self, all_paths, ac, re):
+        '''filter torrents by path (download directory)
+
+        Parameters:
+          all_paths (boolean): whether accept all paths
+          ac (list): accept paths
+          re (list): reject paths
+        '''
+        Filter.__init__(self, all_paths, ac, re)
+
+    def apply(self, torrents):
+        # Pick accepted torrents
+        accepts = set()
+        if self._all: # Accept all torrents (all_paths)
+            accepts = set(torrents)
+        elif len(self._accept) > 0: # Accept specific path torrents (paths)
+            for torrent in torrents:
+                path = torrent.download_dir
+                if path in self._accept or path == '':
```

Comment:
(2) We should not accept a torrent without path unconditionally, so the `path == ''` is inappropriate. This problem should be solved in another way.

Reply:
This filter only support Transmission client for now, and `path == ''` means current is not a Transmission client, so this filter do nothing.

Reply:
I don't agree. Since the torrents in other clients don't have the `download_dir` field, our autoremove-torrents will report an error if we use `path` filter in these clients, so the user will clearly know what happened.

I think it's better to explicitly alert the users instead of sliently ignoring it. The users may take a lot of time to find out what happened.

Reply:
I already added `torrent_obj .download_dir = ''` in other client, so there are no errors here. Maybe I should check all clients API to make this filter work fine with all clients.

Reply:
Just remove `torrent_obj.download_dir = ''` in other clients, so let the autoremove-torrents to report an error for them.

Reply:
ok, I will catch this error and let then know this filter is only work with Transmission.

Reply:
You needn't to catch it by yourself. Just remove torrent_obj.download_dir = '' in other clients. The `task.py` has handled this error.

Reply:
Good, Thanks for your review!

Reply:
Sorry, I have re-checked my code, and it only handles `UnsupportedProperty` for conditions, not for our filters. But don't worry, I will add this check by myself.

Reply:
There is another problem here, I added `disp('download_dir')` to Torrent.__str__, other clients may get an error when printing the Torrent. That's why I add `torrent_obj.download_dir = ''` in other clients 🤣